stats/gRPC.md

```diff
@@ -0,0 +1,125 @@
+# gRPC Stats
+
+Any particular library might provide only a subset of these measures/views/tags.
+Check the language-specific documentation for the list of supported values.
+
+## Units
+
+Units are encoded according to the case-sensitive abbreviations from the [Unified Code for Units of Measure](http://unitsofmeasure.org/ucum.html):
+
+* Latencies are measures in float64 milliseconds, denoted "ms"
+* Sizes are measured in bytes, denoted "By"
+* Dimensionless values have unit "1"
+
+Buckets for distributions in default views are as follows:
+
+* Size in bytes: 0, 1024, 2048, 4096, 16384, 65536, 262144, 1048576, 4194304, 16777216, 67108864, 268435456, 1073741824, 4294967296
+* Latency in ms: 0, 0.01, 0.05, 0.1, 0.3, 0.6, 0.8, 1, 2, 3, 4, 5, 6, 8, 10, 13, 16, 20, 25, 30, 40, 50, 65, 80, 100, 130, 160, 200, 250, 300, 400, 500, 650, 800, 1000, 2000, 5000, 10000, 20000, 50000, 100000
+* Counts (no unit): 0, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192, 16384, 32768, 65536
+
+## Client
+
+### Measures
+
+Client stats are recorded for each individual gRPC request. All measurements except `grpc.io/client/started_count` must be recorded at the end of each gRPC request.
+
+| Measure name                               | Unit | Description                                                                                   |
+|--------------------------------------------|------|-----------------------------------------------------------------------------------------------|
+| grpc.io/client/started_count               |      | Number of all client requests started. Records "1" at the beginning of each gRPC request.     |
+| grpc.io/client/request_count               |      | Number of request messages in a gRPC request.                                                 |
+| grpc.io/client/request_bytes               | By   | Total compressed bytes sent in a gRPC request.                                                |
+| grpc.io/client/uncompressed_request_bytes  | By   | Total uncompressed bytes sent in a gRPC request.                                              |
+| grpc.io/client/response_count              |      | Number of response messages in a gRPC request.                                                |
+| grpc.io/client/response_bytes              | By   | Total compressed bytes received in a gRPC request.                                            |
+| grpc.io/client/uncompressed_response_bytes | By   | Total uncompressed bytes received in a gRPC request.                                          |
+| grpc.io/client/latency                     | ms   | Time between first byte of request received to last byte of response sent, or terminal error. |
+| grpc.io/client/server_latency              | ms   | Propagated from the server and should have the same value as "grpc.io/server/latency".        |
```

Comment:
this is odd, is there a standard gRPC metadata we can read this from? If so, please state this explicitly here. If not, we can't have this measure without more design.

Reply:
Long story short. For the moment no, but we will have it very soon.